R/submission_tmpl.R

```diff
@@ -265,3 +251,34 @@ subset_complete_cases <- function(tmpl_df) {
   )
   tmpl_df[compl_cases, ]
 }
+
+# This function
+switch_get_config <- function(hub_con, config_tasks, hub_path) {
+  input_arg <- rlang::check_exclusive(hub_con, config_tasks, hub_path,
+    .frame = parent.frame()
+  )
+  switch(input_arg,
+    hub_con = {
+      # Signal the deprecation to the user
+      lifecycle::deprecate_warn(
+        "0.11.0",
+        "hubValidations::submission_tmpl(hub_con = )",
+        "hubValidations::submission_tmpl(hub_path = )"
+      )
+      checkmate::assert_class(hub_con, classes = "hub_connection")
+      attr(hub_con, "config_tasks")
+    },
+    config_tasks = {
+      lifecycle::deprecate_warn(
+        "0.11.0",
+        "hubValidations::submission_tmpl(config_tasks = )",
+        "hubValidations::submission_tmpl(hub_path = )"
+      )
+      checkmate::assert_list(config_tasks)
+    },
+    hub_path = {
+      checkmate::assert_directory_exists(hub_path)
+      read_config(hub_path)
```

Comment:
I often hear from modelers that they do not often work directly with the hub on their machine when creating their model outputs. 

If we allow for passing individual config files to be passed to `hub_path` (which could be renamed to `path` to avoid confusion), then modelers could work where they are comfortable working and they could even pass the URL for the config. 

```suggestion
      if (utils::file_test("-d", hub_path)) {
        checkmate::assert_directory_exists(hub_path)
        read_config(hub_path)
      } else {
        checkmate::assert_file_exists(hub_path, extension = "json")
        read_config_file(hub_path)
      }
```